pkg/viewer/viewer.go

```diff
@@ -5,18 +5,89 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-	http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
 
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 package viewer
 
-type handler struct{}
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-pg/pg/v10"
+	"github.com/gofiber/fiber/v2"
+	"k8s.io/klog/v2"
+)
+
+type handler struct {
+	block Block
+}
+
+func NewViewHandler(b Block) handler {
+	return handler{block: b}
+}
+
+func (h *handler) ListBlocks(ctx *fiber.Ctx) error {
+	klog.Infof("viewer ListBlocks")
+
+	arg := BlockArg{
+		From:        ctx.QueryInt("from", 0),
+		Size:        ctx.QueryInt("size", 10),
+		ChannelName: ctx.Query("channel"),
+		StartTime:   int64(ctx.QueryInt("startTime", 0)),
+		EndTime:     int64(ctx.QueryInt("endTime", 0)),
+		BlockNumber: uint64(ctx.QueryInt("blockNumber", 0)),
+		BlockHash:   ctx.Query("blockHash"),
+	}
+	klog.V(5).Infof(" with ctx  %+v arg: %+v\n", *ctx, arg)
+	result, count, err := h.block.List(arg)
+
+	ctx.Set("Content-Type", "application/json")
+	if err != nil {
+		ctx.Status(http.StatusInternalServerError)
+		bs, _ := json.Marshal(map[string]string{"msg": err.Error()})
+		return ctx.Send(bs)
```

Comment:
Is this good enough ? 
https://github.com/bestchains/bc-explorer/blob/main/pkg/listener/handler.go#L61